frontend/src/pages/SignupPage/index.js

```diff
@@ -16,14 +16,64 @@ const STEP_3 = "STEP_3";
 const SignupPage = () => {
   const navigator = useNavigate();
 
-  const [step, setStep] = useState(STEP_1);
-  const [name, setName] = useState("");
-  const [email, setEmail] = useState("");
-  const [selectedInterests, setSelectedInterests] = useState([]);
-  const [addedInterest, setAddedInterest] = useState("");
-  const [error, setError] = useState({ hasError: false, message: "" });
+  const [state, setState] = useState({
+    step: STEP_1,
+    name: "",
+    email: "",
+    selectedInterests: [],
+    addedInterest: "",
+    error: { hasError: false, message: "" },
+  });
 
-  const handleStep = () => {
+  const setError = ({ hasError, message }) => {
+    setState((prevState) => ({
+      ...prevState,
+      error: { hasError: hasError, message: message },
+    }));
+  };
+
+  const setSelectedInterests = (id) => {
+    if (state.selectedInterests.includes(id)) {
+      setState((prevState) => ({
+        ...prevState,
+        selectedInterests: prevState.selectedInterests.filter(
+          (item) => item !== id
+        ),
+      }));
+    } else {
+      setState((prevState) => ({
+        ...prevState,
+        selectedInterests: [...prevState.selectedInterests, id],
+      }));
+    }
+  };
+
+  const handleChange =
+    (input) =>
+    ({ target }) => {
+      if (input === "selectedInterests") {
+        setSelectedInterests(target.id);
+      } else {
+        setState((prevState) => ({ ...prevState, [input]: target.value }));
+      }
+    };
+
+  const nextStep = () => {
+    const { step } = state;
+    const newStep = () => {
+      switch (step) {
+        case STEP_1:
+          return STEP_2;
+        case STEP_2:
+          return STEP_3;
+        default:
+      }
+    };
+    setState((prevState) => ({ ...prevState, step: newStep() }));
+  };
+
+  const Continue = () => {
```

Comment:
I find it slightly confusing to have this function capitalized? We're using camelCase throughout the project. `continueToNextStep`? I know it is verbose, but it's pretty clear what it does, right?

Reply:
I will change it! I wrote it as Continue because that was how they presented in the article. But you're totally right it's inconsistent